src/module/db.ts

```diff
@@ -1,12 +1,136 @@
+import { Game } from '../models/Game';
 
-const DB = new DynamoDB.DocumentClient({
+/**
+ * Custom DynamoDB Response type.
+ */
+type DynamoDBResponse<T> = PromiseResult<T, AWSError>;
+
+/**
+ * Create a DynamoDB DocumentClient instance
+ */
+export const DB = new DynamoDB.DocumentClient({
   apiVersion: '2012-08-10',
   // ////////////////////////////////////
   // Uncomment for local dev only
   // ////////////////////////////////////
-  // region: 'localhost',
-  // endpoint: 'http://localhost:8000',
+  region: 'localhost',
+  endpoint: 'http://localhost:8000',
 });
 
-export default DB;
+/**
+ * Save new connection with connectionId to the ConnectionsTable.
+ * @param {string} connectionId connectionId from event.requestContext
+ */
+export const saveConnection = async (connectionId: string): Promise<DynamoDBResponse<DocumentClient.PutItemOutput>> => {
+  const putParams: DocumentClient.PutItemInput = {
+    TableName: CONNECTIONS_TABLE,
+    Item: {
+      connectionId,
+    },
+  };
+
+  return DB.put(putParams).promise();
+};
+
+/**
+ * Remove connection by connectionId from the ConnectionsTable.
+ * @param {string} connectionId connectionId from event.requestContext
+ */
+export const deleteConnection = async (
+  connectionId: string,
+): Promise<PromiseResult<DocumentClient.DeleteItemOutput, AWSError>> => {
+  const deleteParams: DocumentClient.DeleteItemInput = {
+    TableName: CONNECTIONS_TABLE,
+    Key: {
+      connectionId,
+    },
+  };
+
+  return DB.delete(deleteParams).promise();
+};
+
+/**
+ * Set username for a connection.
+ * @param {string} connectionId connectionId
+ * @param {string} username username
+ */
+export const setUserName = async (
+  connectionId: string,
+  username: string,
+): Promise<DynamoDBResponse<DocumentClient.UpdateItemOutput>> => {
+  const updateParams: DocumentClient.UpdateItemInput = {
+    TableName: CONNECTIONS_TABLE,
+    Key: {
+      connectionId,
+    },
+    ExpressionAttributeNames: { '#usernameKey': 'username' },
+    UpdateExpression: 'set #usernameKey = :usernameVal',
+    ExpressionAttributeValues: {
+      ':usernameVal': username,
+    },
+  };
+
+  return DB.update(updateParams).promise();
+};
+
+/**
+ * Get all connections (rows) from the ConnectionsTable.
+ * Note: only connectionId attribute is retrieved to save DB read cost.
+ */
+export const getAllConnections = async (): Promise<PromiseResult<DocumentClient.ScanOutput, AWSError>> => {
+  const scanParams: DocumentClient.ScanInput = {
+    TableName: CONNECTIONS_TABLE,
+    AttributesToGet: ['connectionId', 'username'], // only get connection Ids from each row
+  };
+
+  return DB.scan(scanParams).promise();
+};
+
+/**
+ * Create a game that takes in a list of connection Ids (users) and write to the GamesTable.
+ * @param {string[]} users an array of users (connectionIds)
+ * @param {string} gameName game name
+ * @param {string} gameType game type
+ * @param {number} gameVersion game version
```

Comment:
game version is also a string!

Reply:
ah ok, but why string though?

Reply:
I'm gonna add BigTwo in here too. gameType can be extended to other games if more games are developed as well. gameVersion would be Japanese/HongKong (depending on what game it is)

Reply:
ic ic